cab-token-generator/java/com/google/auth/credentialaccessboundary/ClientSideCredentialAccessBoundaryFactory.java

```diff
@@ -403,6 +467,70 @@ public void run() {
     }
   }
 
+  /**
+   * Serializes a {@link CredentialAccessBoundary} object into Protobuf wire format.
+   */
+  private byte[] serializeCredentialAccessBoundary(
+      CredentialAccessBoundary credentialAccessBoundary) throws IOException {
+    List<AccessBoundaryRule> rules =
+        credentialAccessBoundary.getAccessBoundaryRules();
+    ClientSideAccessBoundary.Builder accessBoundaryBuilder =
+        ClientSideAccessBoundary.newBuilder();
+
+    for (AccessBoundaryRule rule : rules) {
+      ClientSideAccessBoundaryRule.Builder ruleBuilder =
+          accessBoundaryBuilder.addAccessBoundaryRulesBuilder()
+              .addAllAvailablePermissions(rule.getAvailablePermissions())
+              .setAvailableResource(rule.getAvailableResource());
+
+      if (rule.getAvailabilityCondition() != null) {
```

Comment:
Are there specific conditions that make getAvailabilityCondition nullable? If so, can that be documented?

Reply:
Also, if possible, can you document why availabilityCondition is compiled with CEL and the behavior availability doesn't exist/ isn't compiled?

Reply:
Done. Availability condition is an optional field from users